File: src/main/scala/PatternMatching1.scala
package main.scala

/**
  * Created by ramesh on 20/04/2016.
  */
object PatternMatching1 extends App{
  /**
    * write factorial function with help of pattern matching
    * fact function accepts x of type Int and returns an Int
    * refer to PatternMatching.java for more help
    *
    */

  def fact(x: Int): Int  = x match {
    case 0 => 1
    case _ => x * fact(x-1)

  }

  println(fact(6))
}

File: src/main/java/PatternMatching1.java
package main.java;

/**
 * Created by ramesh on 20/04/2016.
 */
public class PatternMatching1 {

    public static void main(String args[]) {
        System.out.println(fact(6));

    }

    public static Integer fact(Integer x) {
        if (x == 0)
            return 1;
        else
            return x * fact(x - 1);
    }
}
